Lib/http/server.py

```diff
@@ -902,6 +943,30 @@ def guess_type(self, path):
             return guess
         return 'application/octet-stream'
 
+    def parse_range(self):
+        """Return a tuple of (start, end) representing the range header in
+        the HTTP request. If the range header is missing or not resolvable,
+        None is returned. This only supports single part ranges.
+
+        """
+        range_header = self.headers.get('range')
+        if not range_header:
+            return None
+        m = re.match(r'bytes=(\d*)-(\d*)$', range_header)
```

Comment:
From a performance perspective, does it make sense to pre-compile the pattern or to use plain split, startswith and isdigit instead?

Reply:
Have changed to use pre-compiled regex patterns, and IMO it might be a bit more readable to use regex here, and the performance impact of such a simple regular expression should not be an issue here.